### ISSUE TITLE
Migrating to 1.0

### ISSUE THREAD

**Issue by ncfavier** [rust-x-bindings/rust-xcb#265]
I'm trying to update [rust-xcb-util](https://github.com/meh/rust-xcb-util) to the latest version of xcb and running into issues figuring out the new API. For example, the `xcb_window_t`, `xcb_atom_t` etc. types used in [the ffi modules](https://github.com/meh/rust-xcb-util/blob/master/src/ffi/icccm.rs#L111) are no longer exposed. What's the replacement?

**Comment by ncfavier** [comment]
Aha, I didn't expect it to be so simple as replacing `xcb_window_t` with `xcb::x::Window`.

Next question: what is the replacement for `xcb::ReplyError::GenericError(xcb::GenericError { ptr: err })` ([here](https://github.com/meh/rust-xcb-util/blob/39d91e435fe75010f59380dd31f110b0d714574a/src/util.rs#L23))? The functions for resolving an `xcb_generic_error_t` to an `xcb::Error` do not seem to be exposed.

**Comment by rtbo** [comment]
Correct. I can make [`xcb::error::resolve_error`](https://github.com/rust-x-bindings/rust-xcb/blob/5f2359653ae7255c4287667ef9138eb676feb61a/src/error.rs#L108) public.
If the error is emitted by an extension, you need to pass the relevant `ExtensionData`, otherwise the function will panic. The `ExtensionData` array is also private.
I will create the relevant public functions to enable the error error resolution for `xcb-util`.
Do you also need to resolve events ?

**Comment by ncfavier** [comment]
> Do you also need to resolve events ?

No, I don't think so.

**Comment by ncfavier** [comment]
[One more thing](https://www.youtube.com/watch?v=QxBnaMGP2aY): after replacing `xcb_get_property_cookie_t` with `xcb::x::GetPropertyCookie` (and similarly for `xcb_get_property_reply_t`) in the above ffi module, I get the warnings like the following:

```
warning: `extern` block uses type `xcb::x::GetPropertyReply`, which is not FFI-safe
   --> src/ffi/icccm.rs:202:54
    |
202 |     pub fn xcb_icccm_get_wm_protocols_from_reply(reply: *mut xcb::x::GetPropertyReply, protocols: *mut xcb_icccm_get_wm_protocols_reply_t) -> u8;
    |                                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not FFI-safe
    |
    = help: consider adding a `#[repr(C)]` or `#[repr(transparent)]` attribute to this struct
    = note: this struct has unspecified layout
```

Should these types all be marked `repr(C)`?

**Comment by rtbo** [comment]
No. `xcb::x::GetPropertyReply` is a struct containing a pointer to the FFI struct.
What you want is to get the inner pointer.
You can get it with current API through `xcb::Reply::into_raw`, but this will consume the reply, leaving you in charge to free the pointer.
I will add `xcb::Reply::as_raw`, which is more appropriate for your need.
